README.adoc

```diff
@@ -1,5 +1,256 @@
+
+- If we don't have a `factoryExpression` and it is not actorTypeBased, then #self refers to the model.
+
+=== How to reach templateParameters
+
+They are not static and have to be accessible to static helper methods which are called from parallel processes. This can cause issues with parallel building. To void
+this issue the `@ContextAccessor` annotation and a `bindContext` method.
+
+Example:
+[source,java]
+----
+@TemplateHelper
+@ContextAccessor
+public class StoredVariableHelper extends StaticMethodValueResolver {
+
+    public static void bindContext(Map<String, ?> context) {
+        ThreadLocalContextHolder.bindContext(context);
+    }
+
+    public static synchronized String getApiPrefixLocal(Object object) {
+        return (String) ThreadLocalContextHolder.getVariable("apiPrefix");
+    }
+
+    public static synchronized Boolean isGenerateOptionalTypes(Object object) {
+        return Boolean.parseBoolean((String) ThreadLocalContextHolder.getVariable("generateOptionalTypes"));
+    }
+
+    public static synchronized Boolean isGeneratePayloadValidator(Object object) {
+        return Boolean.parseBoolean((String) ThreadLocalContextHolder.getVariable("generatePayloadValidator"));
+    }
+}
+
+----
+
+This only need if we used the templateParameters in a java Helper.
+
+=== Checksum generation
+A checksum is a value calculated from data to detect errors during transmission or storage. It ensures data integrity and any mismatches indicate errors in the data.
+
+* The checksum generation is a goal of the generator-maven-plugin.
+
+Example:
+
+[source,xml]
+----
+                    <execution>
+                        <id>execute-ui-checksum-from-file</id>
+                        <phase>generate-test-resources</phase>
+                        <goals>
+                            <goal>checksum</goal>
+                        </goals>
+                        <configuration>
+                            <ui>${basedir}/src/test/resources/MultiPrincipalTest-ui.model</ui>
+                            <destination>${basedir}/target/test-classes/ui/file</destination>
+                        </configuration>
+                    </execution>
+----
+* The [modelName]ProjectGenerateMojo is responsible for generating the files, while the [modelName]ProjectCalculateChecksumMojo is responsible for generating checksums for the files.
+
+* The .generated-files-[actor]-[actor] contains the actorTypeBased checksums. The .generated-files contains the not actorTypeBased checksums
+
+** A line contains a file name and a md5sum value.
+
+=== Generator ignore file
+
+IMPORTANT: Sometimes a developer needs to replace generated file with custom developed file. On that case the generator has to ignore the given file to keep the edited version. To achieve this `.generator-ignore` file can be used. In this case, every unexpected behavior is the responsibility of the developer.
```

Comment:
> On that case

In that case

Reply:
[4d7240d](https://github.com/BlackBeltTechnology/judo-generator-commons/pull/29/commits/4d7240d5f36e8ab753b70553ed41ed3371f1e13e)